Objects/typeobject.c

```diff
@@ -343,6 +344,49 @@ _PyStaticType_GetBuiltins(void)
 
 /* end static builtin helpers */
 
+static void
+type_set_flags(PyTypeObject *tp, unsigned long flags)
+{
+#ifdef Py_GIL_DISABLED
+    if (tp->tp_flags & Py_TPFLAGS_READY) {
```

Comment:
I think we should just use `FT_ATOMIC_STORE_ULONG_RELAXED`:

* The relaxed atomic store is essentially a normal store, other than the occasional slightly worse code gen we've chatted about.
* In general, with a few exceptions, I don't think these sorts of optimizations are worth the added complexity even if this were an expensive atomic operation.
* In the default build, `FT_ATOMIC_STORE_ULONG_RELAXED(tp->tp_flags, flags)` expands to `tp->tp_flags = flags`

Reply:
Okay, seems reasonable to me.